### PR TITLE
[CLOYSTER-57] Implement fail2ban

### PR DESCRIPTION
**Pull request by lbgracioso**
The name says everything.

### CODE REVIEW

File: src/answerfile.cpp [hunk 1]
```diff
@@ -313,8 +314,24 @@ void AnswerFile::loadNVHPC()
...
 
+void AnswerFile::loadServices() { loadFail2ban(); }
+
+void AnswerFile::loadFail2ban()
+{
+    if (!checkEnabled("fail2ban")) {
+        return;
+    }
+
+    m_services.emplace_back(std::make_shared<fail2ban>());
```
Comment (viniciusferrao) [inline src/answerfile.cpp:325]: I don't see this pointer being used in more than one location. Why is it a `shared_ptr`?

File: include/cloysterhpc/services/IService.h [hunk 2]
```diff
@@ -8,12 +8,14 @@
 
 class IService {
 protected:
+    [[maybe_unused]] virtual void configure() = 0;
+
+public:
+    [[maybe_unused]] virtual void install() = 0;
```
Comment (viniciusferrao) [inline include/cloysterhpc/services/IService.h:14]: If it's a virtual void function does it make sense to be ``[[maybe_unused]]``? I don't get it.

File: include/cloysterhpc/NFS.h [hunk 3]
```diff
@@ -23,7 +23,8 @@ class NFS : public IService {
         const boost::asio::ip::address& address,
         const std::string& permissions);
 
-    void configure();
+    void install() final {};
```
Comment (viniciusferrao) [inline include/cloysterhpc/NFS.h:26]: It does nothing?

File: include/cloysterhpc/answerfile.h [hunk 4]
```diff
@@ -123,6 +127,7 @@ class AnswerFile {
 
     void loadFile(const std::filesystem::path& path);
     std::vector<std::shared_ptr<ITool>> getTools();
+    std::vector<std::shared_ptr<IService>> getServices();
```
Comment (viniciusferrao) [inline include/cloysterhpc/answerfile.h:130]: Same question regarding the `std::shared_ptr`.

File: include/cloysterhpc/services/IService.h (hunk 2 above)
Comment (lbgracioso) [reply include/cloysterhpc/services/IService.h:14]: As an example, the last PR approved that added ITool used install. This one doesn't..

File: include/cloysterhpc/answerfile.h (hunk 4 above)
Comment (lbgracioso) [reply include/cloysterhpc/answerfile.h:130]: I had the feeling we would use it later in different ways, at least when I imagined it and created. For now it's a very simple implementation only to install and configure the tools in a easier way. That's why I put shared.

File: include/cloysterhpc/NFS.h (hunk 3 above)
Comment (viniciusferrao) [reply include/cloysterhpc/NFS.h:26]: But it should do.

**Comment by viniciusferrao** [comment]
I think we can hold this PR a little bit, because to it work correctly we need to have a Firewall infrastructure on the code, and we don't have it.


The Firewall will be probably `firewalld`, however we don't have any control of it yet as we should have.

**Pull request closed without merging**